arch/Kconfig

```diff
@@ -381,6 +381,22 @@ config EXTRA_EXCEPTION_INFO
 
 endmenu # Interrupt configuration
 
+config INIT_ARCH_HW_AT_BOOT
+	bool "Initialize Core Achitecture HW registers at boot"
+	depends on ARCH_SUPPORTS_ARCH_HW_INIT
+	help
+	  This option instructs Zephyr to force the initialization
+	  of architecture core registers during boot to their reset
+	  values. The option is useful when the Zephyr firmware
+	  image is chain-loaded by e.g. a bootloader and we need
+	  to guarantee that the architecture core HW registers
+	  are initialized to their warm reset values (as specified
+	  by the architecture).
+
+	  Note: the functionality is architecture-specific. For the
+	  implementation details refer to each architecture(s) where
+	  the feature is supported.
+
```

Comment:
I think it's not only something architecture specific but at least as well might be SoC-related and thus expand over purely architecture code.

Or do we plan to have yet another similar option for SoC's and boards?

Reply:
Yes, for now this PR concerns only what can be defined as common ARCH-level HW registers. SoCs and Boards have the PLATFORM_INIT Kconfig to set (or reset) specific hardware. So it is not in the scope of this PR.